core/services/relay/evm/mercury/v2/reportcodec/report_codec.go

```diff
@@ -59,58 +59,38 @@ func NewReportCodec(feedID [32]byte, lggr logger.Logger) *ReportCodec {
 	return &ReportCodec{lggr, feedID}
 }
 
-func (r *ReportCodec) BuildReport(paos []reportcodec.ParsedAttributedObservation, f int, validFromTimestamp, expiresAt uint32) (ocrtypes.Report, error) {
-	if len(paos) == 0 {
-		return nil, errors.Errorf("cannot build report from empty attributed observations")
+func (r *ReportCodec) BuildReport(rf reportcodec.ReportFields) (ocrtypes.Report, error) {
+	var merr error
+	if rf.BenchmarkPrice == nil {
+		merr = errors.Join(merr, errors.New("benchmarkPrice may not be nil"))
 	}
-
-	mPaos := reportcodec.Convert(paos)
-
-	timestamp := relaymercury.GetConsensusTimestamp(mPaos)
-
-	benchmarkPrice, err := relaymercury.GetConsensusBenchmarkPrice(mPaos, f)
-	if err != nil {
-		return nil, errors.Wrap(err, "GetConsensusBenchmarkPrice failed")
+	if rf.LinkFee == nil {
+		merr = errors.Join(merr, errors.New("linkFee may not be nil"))
+	} else if rf.LinkFee.Cmp(zero) < 0 {
+		merr = errors.Join(merr, fmt.Errorf("linkFee may not be negative (got: %s)", rf.LinkFee))
 	}
-
-	linkFee, err := relaymercury.GetConsensusLinkFee(mPaos, f)
-	if err != nil {
-		return nil, errors.Wrap(err, "GetConsensusLinkFee failed")
+	if rf.NativeFee == nil {
+		merr = errors.Join(merr, errors.New("nativeFee may not be nil"))
+	} else if rf.NativeFee.Cmp(zero) < 0 {
```

Comment:
Is it worth adding a check to ensure that neither fee can be 0 without the other fee being 0? either both are set or both are 0. Else native being 0 and link being set could have consequences.